--- "..\\php-8.3.8-original\\ext\\phar\\phar.c"	2024-06-04 10:53:17.000000000 -0400
+++ "ext\\phar\\phar.c"	2024-06-05 10:00:36.504356900 -0400
@@ -3302,7 +3302,7 @@
 	if (!file_handle || !file_handle->filename) {
 		return phar_orig_compile_file(file_handle, type);
 	}
-	if (strstr(ZSTR_VAL(file_handle->filename), ".phar") && !strstr(ZSTR_VAL(file_handle->filename), "://")) {
+	if (strstr(ZSTR_VAL(file_handle->filename), ".phar") || strstr(ZSTR_VAL(file_handle->filename), "res://")) {
 		if (SUCCESS == phar_open_from_filename(ZSTR_VAL(file_handle->filename), ZSTR_LEN(file_handle->filename), NULL, 0, 0, &phar, NULL)) {
 			if (phar->is_zip || phar->is_tar) {
 				zend_file_handle f;
