--- "..\\php-8.3.7-original\\ext\\openssl\\config.w32"	2024-05-07 12:35:26.000000000 -0400
+++ "ext\\openssl\\config.w32"	2024-05-12 14:15:57.497013200 -0400
@@ -1,12 +1,12 @@
 // vim:ft=javascript
 
-ARG_WITH("openssl", "OpenSSL support", "no,shared");
+ARG_WITH("openssl", "OpenSSL support", "no");
 
 if (PHP_OPENSSL != "no") {
 	var ret = SETUP_OPENSSL("openssl", PHP_OPENSSL);
 
 	if (ret > 0) {
-		EXTENSION("openssl", "openssl.c xp_ssl.c");
+		EXTENSION("openssl", "openssl.c xp_ssl.c", PHP_OPENSSL_SHARED, "/DZEND_ENABLE_STATIC_TSRMLS_CACHE=1");
 		AC_DEFINE("HAVE_OPENSSL_EXT", PHP_OPENSSL_SHARED ? 0 : 1, "Have openssl");
 		AC_DEFINE("HAVE_OPENSSL", 1);
 	}
